v-next/core/test/internal/tasks/builders.ts

```diff
@@ -0,0 +1,1157 @@
+        });
+      });
+
+      it("should add a named parameter with a description", () => {
+        const builder = new NewTaskDefinitionBuilderImplementation("task-id");
+        const taskAction = () => {};
+        const taskDefinition = builder
+          .setAction(taskAction)
+          .addNamedParameter({
+            name: "param",
+            description: "Parameter description",
+          })
+          .build();
+
+        expectTypeOf(taskDefinition).toEqualTypeOf<NewTaskDefinition>();
+        assert.deepEqual(taskDefinition, {
+          type: TaskDefinitionType.NEW_TASK,
+          id: ["task-id"],
+          description: "",
+          action: taskAction,
+          namedParameters: {
+            param: {
+              name: "param",
+              description: "Parameter description",
+              parameterType: ParameterType.STRING,
+              defaultValue: undefined,
+            },
+          },
+          positionalParameters: [],
+        });
+      });
+
+      it("should add a named parameter with a default value", () => {
+        const builder = new NewTaskDefinitionBuilderImplementation("task-id");
+        const taskAction = () => {};
+        const taskDefinition = builder
+          .setAction(taskAction)
+          .addNamedParameter({
+            name: "param",
+            defaultValue: "default",
+          })
+          .build();
+
+        expectTypeOf(taskDefinition).toEqualTypeOf<NewTaskDefinition>();
+        assert.deepEqual(taskDefinition, {
+          type: TaskDefinitionType.NEW_TASK,
+          id: ["task-id"],
+          description: "",
+          action: taskAction,
+          namedParameters: {
+            param: {
+              name: "param",
+              description: "",
+              parameterType: ParameterType.STRING,
+              defaultValue: "default",
+            },
+          },
+          positionalParameters: [],
+        });
+      });
+
+      it("should add a named parameter with a type", () => {
+        const builder = new NewTaskDefinitionBuilderImplementation("task-id");
+        const taskAction = () => {};
+        const taskDefinition = builder
+          .setAction(taskAction)
+          .addNamedParameter({
+            name: "param",
+            type: ParameterType.INT,
+          })
+          .build();
+
+        expectTypeOf(taskDefinition).toEqualTypeOf<NewTaskDefinition>();
+        assert.deepEqual(taskDefinition, {
+          type: TaskDefinitionType.NEW_TASK,
+          id: ["task-id"],
+          description: "",
+          action: taskAction,
+          namedParameters: {
+            param: {
+              name: "param",
+              description: "",
+              parameterType: ParameterType.INT,
+              defaultValue: undefined,
+            },
+          },
+          positionalParameters: [],
+        });
+      });
+
+      it("should set the default defaultValue to false for when the type is BOOLEAN", () => {
```

Comment:
I think this is worth clarifying, as my TODOs could have been misleading.

Boolean name parameters do have a special treatment, but in the argument parser, not here. 

That special treatment is this:
* If a boolean named parameter has a default value (not all of them have), and if that default value is false, it is consider a "flag".
* Flags can be used in the CLI either like this `--flag true` or just `--flag`.